src/lib/y2network/autoinst/type_detector.rb

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2network/interface_type"
+require "y2network/type_detector"
+require "y2network/sysconfig/interface_file"
+
+module Y2Network
+  module Autoinst
+    # Detects type of given interface based on autoyast profile
+    class TypeDetector < Y2Network::TypeDetector
+      class << self
+        def type_of(iface, section)
+          type_by_sys(iface) || type_by_config(section) || InterfaceType::ETHERNET
```

Comment:
I'm thinking if we should try to do a collision detection here (e.g. device is ethernet according to the /sys but something else according to the profile) and change the precendence (prefer type from AY profile over the /sys one) in such case.

Reply:
well, detection by sys is used mainly for physical interfaces and I am not sure how we can change it. How we can change ib to eth?